### DISCUSSION TITLE
ROFI script not working as expected when launched from i3 config

### DISCUSSION THREAD

**Discussion by i-ideamama** [i3/i3#5593] [Q&A] [answered]
![image](https://github.com/i3/i3/assets/75466610/ade3bf9f-6f6b-4c7e-93c6-2e339d254834)
as shown above, the script launches and works fine when I call it from the terminal

![image](https://github.com/i3/i3/assets/75466610/d3873c53-1318-4237-9b60-9edcb011d973)
However, the above happens then i call it from the i3 config file by a shortcut; none of the options work properly

![image](https://github.com/i3/i3/assets/75466610/3b2a0438-dc43-4016-b343-0279280d7683)
above is the command in my i3 config file that calls the script

![image](https://github.com/i3/i3/assets/75466610/d9811d71-d67f-4992-a43a-c7dfa28bc1e4)
above is the script itself

**NOTE**

1. the above configuration used to work on fedora 37. I am currently running pop-os 22.04 and it isnt working for me
2.  the location where the power.sh script is located is added to path by my bashrc
3.  not only this but another script i made has to same issue

**any help is appreciated**
**thanks**

**Comment by orestisfl** [comment]
> the location where the power.sh script is located is added to path by my bashrc

Try adding it to your `~/.bash_profile` instead.

**Reply by i-ideamama** [reply]
i didnt have a file called `~/.bash_profile` so i created it and added the line `export PATH=$HOME/.local/bin/:$PATH`,.
 But it still isnt working; it just exits when i choose any of the options (same situation is before).
should i have some additional lines in `~/.bash_profile` other than just that one line?

**Reply by orestisfl** [reply, marked as answer]
Okay, from what it seems, for some reason your script is executed with `sh` instead of `bash` since `echo -e` doesn't work evidently from your pictures. I don't understand why but you can test this hypothesis by changing `bindsym .. exec power.sh` with `bindsym .. exec bash -c power.sh` and test again

**Comment by i-ideamama** [comment]
> Okay, from what it seems, for some reason your script is executed with `sh` instead of `bash` since `echo -e` doesn't work evidently from your pictures. I don't understand why but you can test this hypothesis by changing `bindsym .. exec power.sh` with `bindsym .. exec bash -c power.sh` and test again

thanks a lot! this fixed the issue
